### PR TITLE
add weight_norm, wrapper

### PR DESCRIPTION
**Pull request by ArnoldLIULJ**
### Checklist
- [x ] I've tested that my changes are compatible with the latest version of Tensorflow.
- [x ] I've read the [Contribution Guidelines](https://github.com/tensorlayer/tensorlayer/blob/master/CONTRIBUTING.md)
- [x ] I've updated the documentation if necessary.

### CODE REVIEW

File: tensorlayer/layers/weight_norm.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+class WeightNorm(Wrapper):
+    """ This wrapper reparameterizes a layer by decoupling the weight's
+    magnitude and direction. This speeds up convergence by improving the
+    conditioning of the optimization problem.
+    Weight Normalization: A Simple Reparameterization to Accelerate
+    Training of Deep Neural Networks: https://arxiv.org/abs/1602.07868
+    Tim Salimans, Diederik P. Kingma (2016)
+    WeightNorm wrapper works for keras and tf layers.
+
+    Arguments:
```
Comment (zsdonghao) [inline tensorlayer/layers/weight_norm.py]: this is not RST format

File: tensorlayer/layers/weight_norm.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+      layer: a `Layer` instance.
+      in_channels: parameter required when building dynamic layer
+      data_init: If `True` use data dependent variable initialization
+    Raises:
+      ValueError: If not initialized with a `Layer` instance.
+      ValueError: If `Layer` does not contain a `kernel` of weights
+      NotImplementedError: If `data_init` is True and running graph execution
+    
+    Example:
```
Comment (zsdonghao) [inline tensorlayer/layers/weight_norm.py]: this is not RST as well

**Comment by zsdonghao** [comment]
Hi, the docs format is not correct, and I suggest you to add more explanation and examples ~~

**Comment by ArnoldLIULJ** [comment]
Need to rewrite after TL supports layer inside layer feature

**Pull request closed without merging**